src/components/calories/CaloriesForm.js

```diff
@@ -41,27 +42,38 @@ const CaloriesForm = ({ getLowerUpperBound }) => {
           }}
           autoComplete="off"
         >
-          <InputField
-            label="Weight (體重) kg"
+          <Controller
             name="weight"
-            type="number"
-            InputProps={{ inputProps: { min: 0, step: 0.1 } }}
-            {...register('weight', {
-              required: 'Weight (體重) is required.',
-              valueAsNumber: true,
-            })}
-            error={Boolean(errors.weight)}
-            helperText={errors.weight?.message}
+            control={control}
+            rules={validationNumberRegisterOptions}
+            defaultValue=""
+            render={({ field }) => (
+              <InputField
+                {...field}
+                label="Weight (體重) kg"
+                type="number"
+                error={Boolean(errors.weight)}
+                helperText={errors.weight?.message}
+              />
+            )}
           />
-          <SelectField
-            label="Activity Level (活動量)"
+          <Controller
             name="DREFactor"
-            items={activityLevelOptions}
-            {...register('DREFactor', {
-              required: 'Activity Level (活動量) is required.',
-            })}
-            error={Boolean(errors.DREFactor)}
-            helperText={errors.DREFactor?.message}
+            control={control}
+            defaultValue=""
+            rules={{
+              required: 'This field is required (此為必填欄位)',
```

Comment:
Hi, Maybe you can have rules for the valid required fields like `validationNumberRegisterOptions` and use them here. it seems not so consistent, line 48 uses attributes, but here are hardcode the rules.

Reply:
I've made the change! Thanks 👍